deploy/14_faucet.ts

```diff
@@ -1,82 +1,78 @@
-import Web3 from 'web3'
-import networks from '../config/networks'
+declare var global: any;
+import { Contract } from 'ethers';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { DeployFunction } from '@holographxyz/hardhat-deploy-holographed/types';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
 import {
+  hreSplit,
+  genesisDeployHelper,
+  genesisDeriveFutureAddress,
+  generateErc20Config,
   generateInitCode,
-  genesisDeriveFutureAddress, hreSplit,
-  zeroAddress
-} from '../scripts/utils/helpers'
-import { HolographFactory } from '../typechain-types/HolographFactory'
-import { HolographFactoryProxy } from '../typechain-types/HolographFactoryProxy'
-import { HolographRegistry } from '../typechain-types/HolographRegistry'
-import { HolographRegistryProxy } from '../typechain-types/HolographRegistryProxy'
-import { SampleERC20 } from '../typechain-types/SampleERC20'
+} from '../scripts/utils/helpers';
+import { HolographERC20Event, ConfigureEvents } from '../scripts/utils/events';
+import networks from '../config/networks';
 
 const func: DeployFunction = async function (hre1: HardhatRuntimeEnvironment) {
-
-  // Boilerplate
-
-  let { hre, hre2 } = await hreSplit(hre1, global.__companionNetwork)
-  const accounts = await hre.ethers.getSigners()
-  const deployer = accounts[0]
-
-  const network = networks[hre.networkName]
-
-  const web3 = new Web3()
-
-  const error = function (err: string) {
-    hre.deployments.log(err)
-    process.exit()
-  }
-
-  const salt = hre.deploymentSalt
-
-  const holographFactoryProxy = await hre.ethers.getContract<HolographFactoryProxy>('HolographFactoryProxy')
-  const holographFactory = await hre.ethers.getContract<HolographFactory>('HolographFactory')
-  holographFactory.attach(holographFactoryProxy.address)
-
-  const holographRegistryProxy = await hre.ethers.getContract<HolographRegistryProxy>('HolographRegistryProxy')
-  const holographRegistry = await hre.ethers.getContract<HolographRegistry>('HolographRegistry')
-  holographRegistry.attach(holographRegistryProxy.address)
-
-  const chainId = '0x' + network.holographId.toString(16).padStart(8, '0')
-
-  // Get SampleERC20 Contract
-
-  let sampleErc20Address = await (await hre.ethers.getContract<SampleERC20>('SampleERC20')).address
-  if (sampleErc20Address == zeroAddress()) throw 'SampleERC20 is not deployed' // TODO ¯\_(ツ)_/¯
-  hre.deployments.log('reusing "SampleERC20" at:', sampleErc20Address)
-
-  // Deploy Faucet Contract
+  let { hre, hre2 } = await hreSplit(hre1, global.__companionNetwork);
+
+  const accounts = await hre.ethers.getSigners();
+  const deployer: SignerWithAddress = accounts[0];
+
+  const network = networks[hre.networkName];
+
+  const salt = hre.deploymentSalt;
+
+  const holographRegistryProxy = await hre.ethers.getContract('HolographRegistryProxy');
+  const holographRegistry = ((await hre.ethers.getContract('HolographRegistry')) as Contract).attach(
```

Comment:
Is there a reason to remove the correct typings for those contracts?
```
const holographRegistry = await hre.ethers.getContract<HolographRegistry>('HolographRegistry')
```

Reply:
It was a copy paste job from another deploy script. You can try to use correct typings. If it works for you, then can do the same for other deploy scripts in next iteration.